src/db.rs

```diff
@@ -67,19 +63,6 @@ impl ChalkDatabase {
     }
 }
 
-impl ChalkSolveDatabase for ChalkDatabase {
```

Comment:
Is the `self.environment()` method dead code at this point? I would expect so.

Reply:
It's still used in [`chalki`](https://github.com/rust-lang/chalk/blob/master/src/bin/chalki.rs#L199) and a [benchmark](https://github.com/rust-lang/chalk/blob/master/chalk-solve/src/solve/test/bench.rs#L23), but maybe those can be changed...

Reply:
Hmm, actually, that `chalki` use is kind of handy as a debugging tool. Let's leave it as is for now. We might eventually move to allowing you to lower not *the whole program* but *some specific piece of IR* -- but either way the `ToProgramClauses` trait would have to stay public.

Reply:
I think the benchmark however can be altered, we can remove `env` and just do `solve_root_goal(db, ...)`.